### PR TITLE
derivatives restructuring

### LINKED ISSUE #356: Restructuring outputs
Right now if someone runs `fmriprep /bids_directory /outputs participant` they get the following:

```
|-- derivatives
|   `-- sub-387
|       |-- anat
|       |   |-- sub-387_T1w_brainmask.nii.gz
|       |   |-- sub-387_T1w_dtissue.nii.gz
|       |   |-- sub-387_T1w_preproc.nii.gz
|       |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_brainmask.nii.gz
|       |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-CSF_probtissue.nii.gz
|       |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-GM_probtissue.nii.gz
|       |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-WM_probtissue.nii.gz
|       |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_preproc.nii.gz
|       |   |-- sub-387_T1w_target-MNI152NLin2009cAsym_affine.mat
|       |   |-- sub-387_T1w_target-MNI152NLin2009cAsym_warp.nii.gz
|       |   `-- sub-387_T1w_target-meanBOLD_affine.txt
|       `-- func
|           |-- sub-387_task-Beast_run-1_bold_brainmask.nii.gz
|           |-- sub-387_task-Beast_run-1_bold_confounds.tsv
|           |-- sub-387_task-Beast_run-1_bold_preproc.nii.gz
|           |-- sub-387_task-Beast_run-1_bold_space-MNI152NLin2009cAsym_brainmask.nii.gz
|           |-- sub-387_task-Beast_run-1_bold_space-MNI152NLin2009cAsym_preproc.nii.gz
|           `-- sub-387_task-Beast_run-1_bold_target-T1w_affine.txt
|-- log
|   `-- run_workflow
|-- reports
|   `-- sub-387
|       |-- anat
|       |   |-- sub-387_T1w_t1_2_mni.svg
|       |   |-- sub-387_T1w_t1_seg.html
|       |   `-- sub-387_T1w_t1_skull_strip.html
|       `-- func
|           |-- sub-387_task-Beast_run-1_bold_acompcor.html
|           |-- sub-387_task-Beast_run-1_bold_bet_hmc.html
|           |-- sub-387_task-Beast_run-1_bold_flt_bbr.svg
|           `-- sub-387_task-Beast_run-1_bold_tcompcor.html
`-- sub-387.html
```

I would propose refactoring this to the following (with inclusion of recon-all outputs #347 ):

```
|-- freesurfer
|   |-- README
|   |-- pipeline_description.json
|   `-- sub-338
|       |-- label
|       |-- mri
|       |-- scripts
|       |-- stats
|       |-- surf
|       |-- tmp
|       |-- touch
|       `-- trash
`-- fmriprep
    |-- README
    |-- pipeline_description.json
    |-- sub-387
    |   |-- anat
    |   |   |-- sub-387_T1w_brainmask.nii.gz
    |   |   |-- sub-387_T1w_dtissue.nii.gz
    |   |   |-- sub-387_T1w_preproc.nii.gz
    |   |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_brainmask.nii.gz
    |   |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-CSF_probtissue.nii.gz
    |   |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-GM_probtissue.nii.gz
    |   |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_class-WM_probtissue.nii.gz
    |   |   |-- sub-387_T1w_space-MNI152NLin2009cAsym_preproc.nii.gz
    |   |   |-- sub-387_T1w_target-MNI152NLin2009cAsym_affine.mat
    |   |   |-- sub-387_T1w_target-MNI152NLin2009cAsym_warp.nii.gz
    |   |   `-- sub-387_T1w_target-meanBOLD_affine.txt
    |   `-- func
    |       |-- sub-387_task-Beast_run-1_bold_brainmask.nii.gz
    |       |-- sub-387_task-Beast_run-1_bold_confounds.tsv
    |       |-- sub-387_task-Beast_run-1_bold_preproc.nii.gz
    |       |-- sub-387_task-Beast_run-1_bold_space-MNI152NLin2009cAsym_brainmask.nii.gz
    |       |-- sub-387_task-Beast_run-1_bold_space-MNI152NLin2009cAsym_preproc.nii.gz
    |       `-- sub-387_task-Beast_run-1_bold_target-T1w_affine.txt
    `-- sub-387.html
```

Most notable differences:

- Removal of subworkflows (end user does not need them - let's keep them in the ephemeral working directory)
- Removal of `log` subdirectory (the log is actually empty)
- Keeping recon-all and mriqc separate
- Ability to run `fmriprep /bids_directory /bids_directory/derivatives participant` and getting a dataset compatible with [BIDS Derivatives](https://docs.google.com/document/d/1Wwc4A6Mow4ZPPszDIWfCUCRNstn7d_zzaWPcfcHmgI4/edit#heading=h.4xwnp2872y57)

Let me know what do you think.

### PR DESCRIPTION
**Pull request by chrisfilo**
Closes #356.

### CODE REVIEW

File: fmriprep/run_workflow.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ def create_workflow(opts):
         'ignore': opts.ignore,
         'skip_native': opts.skip_native,
         'freesurfer': opts.freesurfer,
+        'reportlets_dir': op.join(op.abspath(opts.work_dir), 'reportlets')
+
```
Comment (effigies) [inline fmriprep/run_workflow.py]: Style: Comma after dict entry, drop blank line.

File: fmriprep/viz/reports.py [hunk 2]
```diff
@@ -162,22 +162,22 @@ def generate_report(self):
...
         dir = root.split('/')[-1]
         try:
             subject = re.search('^(?P<subject_id>sub-[a-zA-Z0-9]+)$', dir).group()
-            out_filename = '{}{}'.format(subject, '.html')
-            report = Report(root, config, out_dir, run_uuid, errno, out_filename)
-            report.generate_report()
-        except AttributeError as e:
+        except AttributeError:
             continue
+        out_filename = '{}{}'.format(subject, '.html')
```
Comment (effigies) [inline fmriprep/viz/reports.py]: Could just be `out_filename = '{}.html'.format(subject)`

File: fmriprep/workflows/base.py [hunk 3]
```diff
@@ -34,15 +34,15 @@ def base_workflow_enumerator(subject_list, task_id, settings, run_uuid):
...
+        fsdir.inputs.derivatives = os.path.join(settings['output_dir'])
 
     generated_list = []
     for subject in subject_list:
         generated_workflow = base_workflow_generator(subject, task_id=task_id,
                                                      settings=settings)
         if generated_workflow:
             generated_workflow.config['execution']['crashdump_dir'] = (
-                os.path.join(settings['output_dir'], 'log', subject, run_uuid)
+                os.path.join(settings['output_dir'], "fmriprep", "sub-" + subject, 'log', subject, run_uuid)
```
Comment (effigies) [inline fmriprep/workflows/base.py]: Does this work correctly when you specify `--participant-label sub-01` (or similar), or do you end up with `sub-sub-01`?

Comment (chrisfilo) [inline fmriprep/workflows/base.py]: It would not, but the `--participant-label` is not intended to include `sub-` prefix.

Comment (effigies) [inline fmriprep/workflows/base.py]: Ah. Well, it works very well with it. Good to know.

**Pull request merged**